docs/guide/why.md

```diff
@@ -51,7 +51,9 @@ Ensuring optimal output and behavioral consistency between the dev server and th
 
 ## Why Not Bundle with esbuild?
 
-While `esbuild` is extremely fast and is already a very capable bundler for libraries, some of the important features needed for bundling _applications_ are still work in progress - in particular code-splitting and CSS handling. For the time being, Rollup is more mature and flexible in these regards. That said, we won't rule out the possibility of using `esbuild` for production builds when it stabilizes these features in the future.
+Vite's adoption of Rollup's plugin API and infrastructure heavily contributed to its success in the ecosystem, however, it isn't compatible with `esbuild`'s API. For the time being, we believe that Rollup offers a better performance-vs-flexibility tradeoff.
+
+That said, `esbuild` has progressed a lot in the past years, and we won't rule out the possibility of using `esbuild` for production builds in the future. We will keep taking advantage of new capabilities as they are released, as we have done with JS and CSS minification where `esbuild` allowed Vite to get a performance boost while avoiding disruption for its ecosystem.
```

Comment:
I removed the history bit of `esbuild` as it's not the main reason to the question anymore. I hope the first paragraph now directly reflects why we didn't go with esbuild.